src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -202,6 +202,68 @@ public bool NoDownloadRequested(ConnectNewHeadersResult connectNewHeadersResult)
                 return (connectNewHeadersResult.DownloadTo == null)
                        && (connectNewHeadersResult.DownloadFrom == null);
             }
+
+            /// <summary>
+            /// Initial setup for tests 18-20, 28.
+            /// Chain header tree setup. Initial chain has 4 headers.
+            /// SetUp:
+            ///                        =8d=9d=10d
+            ///                   6a=7a=8a=9a
+            /// GENESIS=1=2=3=4=5=
+            ///                   6b=7b=8b=9b
+            ///             3c=4c=5c
+            /// </summary>
+            /// <param name="initialChainSize">Initial chain size.</param>
+            /// <param name="cht">ChainHeaderTree.</param>
+            /// <param name="initialChainTip">Initial chain tip.</param>
+            /// <param name="chainATip">out Chain A tip.</param>
```

Comment:
I don't think we need these out parameters at all, we do have extension methods providing those headers already. I think it is called GetPeerTipChainedHeaderByPeerId.